fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/search/LuceneOptimizedIndexSearcher.java

```diff
@@ -110,5 +116,41 @@ protected void search(List<LeafReaderContext> leaves, Weight weight, Collector c
         }
     }
 
+    /**
+     * This overridden call will attempt to cache the relevant terms low level blocks in the case where we need scores.
+     *
+     * @param query query implementation
+     * @param scoreMode mode for scoring
+     * @param boost boost for scores
+     * @return Weight of query
+     * @throws IOException
+     */
+    @Override
+    public Weight createWeight(final Query query, final ScoreMode scoreMode, final float boost) throws IOException {
+        if (scoreMode.needsScores() && getExecutor() != null) {
+            List<Term> terms = getQueriesTerms(query);
+            for (Term term : terms) {
+                for (final LeafReaderContext ctx : getTopReaderContext().leaves()) {
+                    // Do not block on these pre-fetches...
+                    CompletableFuture.runAsync(() -> {
+                        try {
+                            cacheTermsEnum(ctx, term);
+                        } catch (Exception e) {
+                            CompletableFuture.failedFuture(e);
```

Comment:
Given that this future just gets swallowed, I think we should probably just make this a functionally-empty `catch` block. (I think one of our style checkers gets mad if you make the `catch` block actually empty, but maybe just leaving a comment that we're swallowing the error is sufficient here).

Reply:
agreed, fixing...